src/librustc_resolve/lib.rs

```diff
@@ -1861,7 +1860,7 @@ impl<'a> Resolver<'a> {
                                 }
                                 TraitItemKind::Method(ref sig, _) => {
                                     let type_parameters =
-                                        HasTypeParameters(&sig.generics,
+                                        HasTypeParameters(&trait_item.generics,
```

Comment:
I think we'll need to add this type parameter rib to each variant in this match statement because each `HasTypeParamters` will need a different rib kind. Since nothing other than methods has generics right now, would it work to save that change until we are actually implementing associated type generics?

(this applies to both trait items and impl items)

Reply:
Well, it depends. I would think we just want to rename `MethodRibKind` to something else like `TraitOrImplItemRibKind`, but it might be important to distinguish those cases for error messages? I kinda' doubt it, but have to look around.